app/services/unpublish_service.rb

```diff
@@ -88,6 +86,21 @@ def remove_and_redirect(edition, redirect_path, explanatory_note: nil)
 
 private
 
+  def create_withdrawal(edition, public_explanation)
+    if edition.withdrawn?
+      Withdrawal.create!(
+        public_explanation: public_explanation,
+        published_status: edition.status.details.published_status,
```

Comment:
Would it be more consistent to use delegation here? `edition.published_status` has a certain ring to it ;-).

Reply:
Actually, I see how that wouldn't make much sense outside of the context of a withdrawal. Perhaps this would read more clearly if `edition.status` were clarified to be a withdrawal?

Reply:
The if statement suggests the edition status is withdrawn however if this is not clear i could assign edition.status to a variable e.g
```
if edition.withdrawn?
withdrawal = edition.status
Withdrawal.create!(
        public_explanation: public_explanation,
        published_status: withdrawal.details.published_status
``` ?

Reply:
👍